### PR TITLE
Initial Docusaurus Setup

### PR DESCRIPTION
**Pull request by Lewiky**
Setup Docusaurus for documentation and write a few basic documentation pages

### CODE REVIEW

File: TempleEight/README.md [hunk 1]
```diff
@@ -0,0 +1,33 @@
+# Website
+
+This website is built using [Docusaurus 2](https://v2.docusaurus.io/), a modern static website generator.
+
+### Installation
+
+```
```
Comment (xsanda) [inline TempleEight/README.md]: worth stating this must be run in the relevant folder?

File: TempleEight/blog/2019-05-30-welcome.md [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+id: welcome
+title: Welcome
+author: Yangshun Tay
+author_title: Front End Engineer @ Facebook
+author_url: https://github.com/yangshun
+author_image_url: https://avatars0.githubusercontent.com/u/1315101?s=400&v=4
+tags: [facebook, hello, docusaurus]
+---
+
+Blog features are powered by the blog plugin. Simple add files to the `blog` directory. It supports tags as well!
```
Comment (xsanda) [inline TempleEight/blog/2019-05-30-welcome.md]: Do we want this file?

File: TempleEight/docs/templefile-spec.md [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+
+# TempleFile DSL Specification
+
+## Primitives
+
+```
+bool
+string(maxLength: int, minLength: int)
+int(max: int, min: int = 0)
+float(max: float, min: float = 0.0f, precision: int = 8)
```
Comment (xsanda) [inline TempleEight/docs/templefile-spec.md]: 0.0 plz

File: TempleEight/static/img/undraw_docusaurus_tree.svg [hunk 4]
```diff
@@ -0,0 +1 @@
...84463,3.60248,3.60248,0,0,1,.06636-.65027,3.82638,3.82638,0,0,1,.18447-.62373,3.48858,3.48858,0,0,1,.30656-.57064,3.197,3.197,0,0,1,.91436-.91568,3.44685,3.44685,0,0,1,.572-.30523,3.344,3.344,0,0,1,.62372-.18578,3.06907,3.06907,0,0,1,1.30053,0,3.22332,3.22332,0,0,1,1.19436.491,5.02835,5.02835,0,0,1,.50429.41139,4.8801,4.8801,0,0,1,.41139.50429,3.38246,3.38246,0,0,1,.30522.57064,3.47806,3.47806,0,0,1,.25215,1.274A3.36394,3.36394,0,0,1,426.36,446.865a5.02708,5.02708,0,0,1-.50429.4114,3.3057,3.3057,0,0,1-1.84463.55737m26.54143-1.65884a3.38754,3.38754,0,0,1-2.35024-.96877,5.04185,5.04185,0,0,1-.41007-.50428,3.27532,3.27532,0,0,1-.55737-1.84463,3.38659,3.38659,0,0,1,.96744-2.34892,5.02559,5.02559,0,0,1,.50429-.41139,3.44685,3.44685,0,0,1,.572-.30523,3.3432,3.3432,0,0,1,.62373-.18579,3.06952,3.06952,0,0,1,1.30052,0,3.22356,3.22356,0,0,1,1.19436.491,5.02559,5.02559,0,0,1,.50429.41139,3.38792,3.38792,0,0,1,.96876,2.34892,3.72635,3.72635,0,0,1-.06636.65026,3.37387,3.37387,0,0,1-.18579.62373,4.71469,4.71469,0,0,1-.30522.57064,4.8801,4.8801,0,0,1-.41139.50429,5.02559,5.02559,0,0,1-.50429.41139,3.30547,3.30547,0,0,1-1.84463.55737" transform="translate(-35.5 -118.5)" fill-rule="evenodd"/></svg>
```
Comment (xsanda) [inline TempleEight/static/img/undraw_docusaurus_tree.svg]: oh god why

Comment (Lewiky) [reply TempleEight/static/img/undraw_docusaurus_tree.svg]: They need to be in the github-pages.... as such have to be in the repo sorry ;_;

... Lewis

File: TempleEight/blog/2019-05-29-hello-world.md [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+tags: [hello, docusaurus]
+---
+
+Welcome to this blog. This blog is created with [**Docusaurus 2 alpha**](https://v2.docusaurus.io/).
+
+<!--truncate-->
+
+This is a test post.
+
+A whole bunch of other information.
```
Comment (jaylees14) [inline TempleEight/blog/2019-05-29-hello-world.md]: Could we just have 1 of the example blogs if they're not that relevant?

File: TempleEight/docs/primatives.md [hunk 6]
```diff
@@ -0,0 +1,103 @@
...
+---
+
+TempleFile supports 8 primative types, roughly aligned with those found in `SQL` style languages. Any property of a service can be of any of these types. 
+
+```temple
+bool
+string(maxLength: int, minLength: int)
+int(max: int, min: int = 0)
+float(max: float, min: float = 0.0f, precision: int = 8)
+date
```
Comment (jaylees14) [inline TempleEight/docs/primatives.md]: I believe the spelling is `primitive` not `primative`

File: TempleEight/docusaurus.config.js [hunk 7]
```diff
@@ -0,0 +1,91 @@
+module.exports = {
+  title: 'Temple',
+  tagline: 'makes it easy to get new microservices up and running, at lightning speed',
+  url: 'https://your-docusaurus-test-site.com',
```
Comment (jaylees14) [inline TempleEight/docusaurus.config.js]: `https://templeeight.github.io/temple-docs/`

**Comment by jaylees14** [comment]
Just a thought - is there a Dockerfile included with this? Could be useful to test locally without the dependencies

**Pull request merged**